datafusion/core/tests/sqllogictests/src/engines/postgres/mod.rs

```diff
@@ -290,27 +249,20 @@ impl sqllogictest::AsyncDB for Postgres {
             return Ok(DBOutput::StatementComplete(0));
         }
         let rows = self.client.query(sql, &[]).await?;
-        let output = rows
-            .iter()
-            .map(|row| {
-                row.columns()
-                    .iter()
-                    .enumerate()
-                    .map(|(idx, column)| cell_to_string(row, column, idx))
-                    .collect::<Vec<String>>()
-            })
-            .collect::<Vec<_>>();
 
-        if output.is_empty() {
-            let stmt = self.client.prepare(sql).await?;
-            Ok(DBOutput::Rows {
-                types: vec![DFColumnType::Any; stmt.columns().len()],
-                rows: vec![],
-            })
+        if rows.is_empty() {
+            return Ok(DBOutput::StatementComplete(0));
```

Comment:
Datafusion currently considers empty results as "statements" https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sqllogictests/src/engines/datafusion/normalize.rs#L32-L35.
This change makes Postgres produce the same result when there are no values.

Technically, an opposite change is possible: update Datafusion to consider an empty batch as empty query results, but with types.

Reply:
The opposite (empty result as a query result with schema) is probably a better behavior as it would allow checking types for statements like `COUNT(*) FROM foo WHERE false`

Reply:
Makes sense. I updated the implementation here https://github.com/apache/arrow-datafusion/pull/5253/commits/e9733d76179f2a9cbb841d94f6239c12ab7115b8.

As a side note, I noticed that `sqllogictest-rs` doesn't implement checks when the expected record is a statement, for example, `statement ok`. In this case, even if the client implementation returns results, neither types nor actual rows would be checked.